doc/dash/smart-switch-ip-address-assigment/smart-switch-ip-address-assignment.md

```diff
@@ -0,0 +1,372 @@
+
+```yang
+    container sonic-device_metadata {
+
+        container DEVICE_METADATA {
+            leaf type {
+                    type string {
+                        length 1..255;
+                        pattern "<All existing types>|SmartSwitch";
+                    }
+                }
+        }
+    }
+```
+
+##### MID_PLANE_BRIDGE and DPUS tables #####
+
+```yang
+    container sonic-smart-switch {
+
+        container MID_PLANE_BRIDGE {
+
+            description "MID_PLANE_BRIDGE part of config_db.json";
+
+            container GLOBAL {
+                leaf bridge {
+                    type string {
+                        pattern "bridge_midplane";
+                    }
+                    description "Name of the midplane bridge";
+
+                    must "(current()/../ip_prefix) and (current()/../members)";
+                }
+
+                leaf ip_prefix {
+                    type inet:ipv4-prefix;
+                    description "IP prefix of the midplane bridge";
+                }
+
+                leaf-list members {
+                    description "List of the DPU interfaces.";
+
+                    type string {
+                        pattern "Dpu[0-9]+";
+                    }
+                }
+            }
+            /* end of container GLOBAL */
+        }
+        /* end of container MID_PLANE_BRIDGE */
+
+        container DPUS {
+            description "DPUS part of config_db.json";
+
+            list DPUS_LIST {
+                key "dpu_name";
+
+                leaf dpu_name {
+                    description "Name of the DPU";
+                    type string {
+                        pattern "DPU[0-9]+";
+                    }
+                }
+
+                leaf interface {
+                    description "Name of the interface that represents DPU";
+
+                    type string {
+                        pattern "Dpu[0-9]+";
+                    }
+                }
+            }
+            /* end of container DPUS_LIST */
+        }
+        /* end of container DPUS */
+    }
+    /* end of container sonic-smart-switch */
+
+```
+
+## Warmboot and Fastboot Design Impact ##
+
+The feature has no requirements for a warm and fast boot.
+
+## Memory Consumption ##
+
+The feature has minimal impact on the memory consumption. Overall it requires less than 1KB of disk and RAM space to store the new configuration.
+
+## Restrictions/Limitations ##
+
+## Testing Requirements/Design ##
+
+### Unit Test cases ###
+
+1. Add new YANG model tests for the new MID_PLANE_BRIDGE and DPUS table.
+2. Extend existing YANG model tests to cover MGMT_INTERFACE table changes.
+3. Add new tests to cover the handling of MID_PLANE_BRIDGE and DPUS tables in sonic-cfggen utility.
+4. Add new tests to cover DHCP server configuration generation for the Smart Switch in sonic-cfggen utility.
+5. Extend interface-config.service tests to verify midplane bridge configuration generation.
```

Comment:
I think we should have a test to ensure these midplane IPs are not advertised via BGP